zuul-core/src/main/java/com/netflix/zuul/message/http/HttpQueryParams.java

```diff
@@ -72,12 +78,17 @@ public static HttpQueryParams parse(String queryString) {
                 String name = s.substring(0, i);
                 String value = s.substring(i + 1);
 
-                try {
-                    name = URLDecoder.decode(name, "UTF-8");
-                    value = URLDecoder.decode(value, "UTF-8");
-                }
-                catch (Exception e) {
-                    // do nothing
+                if (!propertyFactory.getBooleanProperty(
+                    ZuulConstants.ZUUL_KEEP_ORIGINAL_QUERY_STRING_ENCODING,
```

Comment:
I don't think this is the correct thing to do.   Preserving the encoding is fine, but the name and value still need to be validated.  If this flag is off, all that needs to happen is the return value of URLDecoder should be discarded.